### PR TITLE
[SQL] implement ARRAY_REMOVE function

### PR DESCRIPTION
**Pull request by abhizer**
Is this a user-visible change (yes/no): yes

### CODE REVIEW

File: docs/sql/array.md [hunk 1]
```diff
@@ -43,6 +43,7 @@ as `CREATE TABLE data AS (CITIES VARCHAR ARRAY, COUNTRY VARCHAR)`.
...
 | `ARRAY_LENGTH(` _array_ `)`                                          | Another name for `CARDINALITY`                                                                                                                                                                                                | `ARRAY_LENGTH( ARRAY [1, 2, 3])` => `3`                             |
 | `ARRAY_POSITION(` _array_ `, ` _element_ `)`                         | Returns the (1-based) index of the first _element_ of the _array_ as a long. Returns 0 if _element_ doesn't exist in _array_. If the _element_ is `NULL`, `NULL` is returned.                                                 | `ARRAY_POSITION(ARRAY [1, 3, 4, 6], 4)` => 3                        |
+| `ARRAY_REMOVE(` _array_ `, ` _element_ `)`                           | Removes all elements that are equal to _element_ from the array                                                                                                                                                               | `ARRAY_REMOVE(ARRAY [2, 2, 6, 8], 2)` => `[6, 8]`                   |
```
Comment (mihaibudiu) [inline docs/sql/array.md:46]: wasn't there a funny behavior for nulls?

File: sql-to-dbsp-compiler/SQL-compiler/src/test/java/org/dbsp/sqlCompiler/compiler/sql/simple/ArrayFunctionsTests.java [hunk 2]
```diff
@@ -107,6 +107,54 @@ public void testArrayPositionDiffTypes() {
         );
     }
 
+    @Test
```
Comment (mihaibudiu) [inline sql-to-dbsp-compiler/SQL-compiler/src/test/java/org/dbsp/sqlCompiler/compiler/sql/simple/ArrayFunctionsTests.java:110]: is there a test for a null array?
Please note that NULL and CAST(NULL AS INTEGER ARRAY) are not the same.

**Review by mihaibudiu — CHANGES_REQUESTED** [review]

File: docs/sql/array.md (hunk 1 above)
Comment (abhizer) [reply docs/sql/array.md:46]: For element as null, we return null. Same for CAST(null as integer array).

File: sql-to-dbsp-compiler/SQL-compiler/src/test/java/org/dbsp/sqlCompiler/compiler/sql/simple/ArrayFunctionsTests.java (hunk 2 above)
Comment (abhizer) [reply sql-to-dbsp-compiler/SQL-compiler/src/test/java/org/dbsp/sqlCompiler/compiler/sql/simple/ArrayFunctionsTests.java:110]: No I will add the test.

File: docs/sql/array.md (hunk 1 above)
Comment (mihaibudiu) [reply docs/sql/array.md:46]: yes, but this behavior has to be documented

File: docs/sql/array.md [hunk 3]
```diff
@@ -42,7 +42,8 @@ as `CREATE TABLE data AS (CITIES VARCHAR ARRAY, COUNTRY VARCHAR)`.
...
 | `ARRAY_LENGTH(` _array_ `)`                                          | Another name for `CARDINALITY`                                                                                                                                                                                                | `ARRAY_LENGTH( ARRAY [1, 2, 3])` => `3`                             |
-| `ARRAY_POSITION(` _array_ `, ` _element_ `)`                         | Returns the (1-based) index of the first _element_ of the _array_ as a long. Returns 0 if _element_ doesn't exist in _array_. If the _element_ is `NULL`, `NULL` is returned.                                                 | `ARRAY_POSITION(ARRAY [1, 3, 4, 6], 4)` => 3                        |
+| `ARRAY_POSITION(` _array_ `, ` _element_ `)`                         | Returns `NULL` if _array_ or _element_ is `NULL`. Returns the (1-based) index of the first _element_ of the _array_ as a long. Returns 0 if _element_ doesn't exist in _array_.                                               | `ARRAY_POSITION(ARRAY [1, 3, 4, 6], 4)` => 3                        |
```
Comment (mihaibudiu) [inline docs/sql/array.md:45]: there were some recent changes merged in Calcite which will change the behavior of these functions when the array argument is a NULL literal without a cast. But I don't think it's important enough to explain this.

**Review by mihaibudiu — APPROVED** [review]

**Pull request merged**